Source: Timo Gurr <timo.gurr@gmail.com>
Upstream: Reported, https://bugs.kde.org/show_bug.cgi?id=350915
Reason: Fix build with jpeg9

diff -Naur digikam-4.9.0/core/libs/dimg/loaders/jpegloader.cpp digikam-4.9.0.new/core/libs/dimg/loaders/jpegloader.cpp
--- digikam-4.9.0/core/libs/dimg/loaders/jpegloader.cpp	2015-04-06 18:44:29.000000000 +0200
+++ digikam-4.9.0.new/core/libs/dimg/loaders/jpegloader.cpp	2015-07-31 14:59:16.313564425 +0200
@@ -250,7 +250,7 @@
     }
 
     // read image information
-    jpeg_read_header(&cinfo, true);
+    jpeg_read_header(&cinfo, boolean(true));
 
     // read dimension (nominal values from header)
     int w = cinfo.image_width;
@@ -300,8 +300,8 @@
     if (m_loadFlags & LoadImageData)
     {
         // set decompression parameters
-        cinfo.do_fancy_upsampling = false;
-        cinfo.do_block_smoothing  = false;
+        cinfo.do_fancy_upsampling = boolean(false);
+        cinfo.do_block_smoothing  = boolean(false);
 
         // handle scaled loading
         if (scaledLoadingSize)
@@ -784,8 +784,8 @@
         }
     }
 
-    jpeg_set_quality(&cinfo, quality, true);
-    jpeg_start_compress(&cinfo, true);
+    jpeg_set_quality(&cinfo, quality, boolean(true));
+    jpeg_start_compress(&cinfo, boolean(true));
 
     kDebug() << "Using LibJPEG quality compression value: " << quality;
 
